### DISCUSSION TITLE
Why am I getting a FileNotFoundException when instancing a WixSharp UI Dialog?

### DISCUSSION THREAD

**Discussion by NJullienSweet** [oleg-shilo/wixsharp#1642] [Q&A] [answered]
I am getting a `FileNotFoundException` each time I try to open a dialog inheriting from `ManagedForm`, from namespace `WixSharp.UI`.
I tried and succeeded opening open basic WinForms or WPF windows, but it appears I can't with `ManagedForm`.
![image](https://github.com/user-attachments/assets/c9ce403c-2afe-4d6b-a245-743571108aff)
Translation:
> Could not load the file or assembly 'WixSharp.UI, Version=2.3.2.0, Culture=neutral, PublicKeyToken=3775edd25acc43c2' or one of its dependencies. The specified file cannot be found.

Here is the stripped out BA code:

```csharp
[assembly: BootstrapperApplicationFactory(typeof(WixBA.WixBAFactory))]
namespace WixBA
{
	public class WixBAFactory : BaseBootstrapperApplicationFactory
	{
		protected override mba.IBootstrapperApplication Create(IEngine engine, IBootstrapperCommand command)
		{
			return new MyBootstrapperApplication(engine, command);
		}
	}
}

public class MyBootstrapperApplication : mba.BootstrapperApplication
{
	public IEngine Engine => engine;
	private Form _currentForm;

	public MyBootstrapperApplication(IEngine engine, IBootstrapperCommand command) : base(engine) { }

	protected override void Run()
	{
		// ...

		try
		{
			ShowDialog();
		}
		catch (Exception ex)
		{
			MessageBox.Show(ex.Message, "Une erreur est survenue", MessageBoxButtons.OK, MessageBoxIcon.Error);
		}

		engine.Detect();
	}

	private void ShowDialog()
	{
		_currentForm = new WixSharp_UI_Dialog1();
		_currentForm.ShowDialog();
	}
}
```

The `WixSharp_UI_Dialog1` class is the raw template from the VS extension
![image](https://github.com/user-attachments/assets/26a6bd5d-b9c5-4b6a-bc69-1cd37672857f)

**Comment by oleg-shilo** [comment, marked as answer]
WixSharp UI dialog is designed to be hosted by the MSI runtime within so-called EmbeddedUI (assembly implementing WixSharp dialog).
You cannot show the instance of the dialog that you created by yourself. 

If you simply want to show somedialog then you need to use `new Form(...)`
